### DISCUSSION TITLE
Error when pulling on Termux

### DISCUSSION THREAD

**Discussion by kud** [marlonrichert/zsh-snap#210] [General]
When I use any pull commands, it gives me this error:

```
❯ znap pull
From https://github.com/marlonrichert/zsh-autocomplete
 * branch            main       -> FETCH_HEAD
BUG: run-command.c:1525: you must provide a non-zero number of processes!
From https://github.com/zsh-users/zsh-autosuggestions
 * branch            master     -> FETCH_HEAD
BUG: run-command.c:1525: you must provide a non-zero number of processes!
From https://github.com/zsh-users/zsh-syntax-highlighting
 * branch            master     -> FETCH_HEAD
BUG: run-command.c:1525: you must provide a non-zero number of processes!
```

Weird thing is: I don't have this issue when I git pull on my own. Another weird thing is this error does not change anything, I mean it works anyway.

I've got this on Termux.

**Comment by michael-odell** [comment]
I'm seeing the same errors, although I'm not using Termux.  I see it independently of terminal.  I'm on macOS and tried it in Terminal, iTerm2, and tmux.

It seems to be related to the `--jobs=0` argument znap is passing to `git pull`.  I'm not sure what this is supposed to signify -- maybe to run _everything_ in parallel with no cap.  But I have worked around it for now by removing `$jopt` from the `git pull` command in `functions/.znap.pull`.  This makes the problem go away, but probably also disables parallel pulls.  I have few enough repos being pulled that this is not terribly onerous for me.

I see suggestions in [this stackoverflow question](https://stackoverflow.com/questions/74783565/git-error-when-pulling-from-origin-you-must-provide-a-non-zero-number-of-proce) of two not-quite contradictory things -- that it's [an intentional change](https://github.com/git/git/commit/51243f9f0f6a932ea579fd6f8014b348f8c2a523) but also that it must be a bug given the way it's reporting.

I'm using git version `2.39.1` from homebrew.  Hopefully a git update coming soon fixes it?  Or maybe znap could be tweaked to be configurable so that it doesn't pass this option as a stopgap?

**Reply by kud** [reply]
Thank you @michael-odell  😌

**Reply by michael-odell** [reply]
Glad I could help!  Thanks for posting the question @kud .  When weird things like this crop up it's always nice to have confirmation it's not _just me_ seeing it.

**Reply by kud** [reply]
That's right yeah 😊

**Reply by marlonrichert** [reply]
This is now fixed with ea769c70f7aaf00f438739cbeaddf88b073f5017.

**Reply by michael-odell** [reply]
Yup, that works here, too.  Thanks @marlonrichert !

**Reply by marlonrichert** [reply]
No problem! 😊

PS: If you enjoy using my software, please consider [sponsoring me](https://github.com/sponsors/marlonrichert). 💝
